hs_core/views/__init__.py

```diff
@@ -662,43 +702,17 @@ def delete_multiple_files(request, shortkey, *args, **kwargs):
 
 def delete_resource(request, shortkey, *args, **kwargs):
     res, _, user = authorize(request, shortkey, needed_permission=ACTION_TO_AUTHORIZE.DELETE_RESOURCE)
-
     res_title = res.metadata.title
-    res_id = shortkey
     res_type = res.resource_type
-    resource_related_collections = [col for col in res.collections.all()]
-    owners_list = [owner for owner in res.raccess.owners.all()]
-    ajax_response_data = {'status': 'success'}
-    try:
-        hydroshare.delete_resource(shortkey)
-    except ValidationError as ex:
-        if request.is_ajax():
-            ajax_response_data['status'] = 'error'
-            ajax_response_data['message'] = str(ex)
-            return JsonResponse(ajax_response_data)
-        else:
-            request.session['validation_error'] = str(ex)
-            return HttpResponseRedirect(request.META['HTTP_REFERER'])
-
-    # if the deleted resource is part of any collection resource, then for each of those collection
-    # create a CollectionDeletedResource object which can then be used to list collection deleted
-    # resources on collection resource landing page
-    for collection_res in resource_related_collections:
-        o=CollectionDeletedResource.objects.create(
-             resource_title=res_title,
-             deleted_by=user,
-             resource_id=res_id,
-             resource_type=res_type,
-             collection=collection_res
-             )
-        o.resource_owners.add(*owners_list)
-
+    task = delete_resource_task.apply_async((shortkey, request.user.username))
+    task_id = task.task_id
+    task_dict = get_task_by_id(task_id, name='resource delete', payload=shortkey)
+    create_task_notification(task_id, name='resource delete', payload=shortkey)
     post_delete_resource.send(sender=type(res), request=request, user=user,
```

Comment:
Should this be in the task?  We shouldn't be sending the post_delete signal before the delete completes.

Reply:
Good catch. I need to fix this.

Reply:
fixed. Check out my latest commit. I cannot put this post_delete_resource.send into the resource delete task since the request object cannot be passed to the task due to serialization problem. So I find another way to do it.